bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/AbstractExpression.java

```diff
@@ -240,11 +272,36 @@ public Date getFinalFireTime() {
             }
         }
 
-        if (getCandidates().isEmpty()) {
-            return null;
+        Date lastCandidate = null;
+
+        if (!getCandidates().isEmpty()) {
+            if (getCandidates().size() == 1) {
+                lastCandidate = getCandidates().get(0);
+            } else {
+
+                while (getCandidates().size() == maximumCandidates) {
```

Comment:
I am not sure I fully understand your comment here. For cron the overall number of candidates can be limited (as per Quartz definition), but for practical reasons, in case there are unlimited number of candidates (like in your infamous example), the YEARS parameters is limited to 2100 anyways. So, infinite series are finite from a practical point of view.